### PR TITLE
Create draft BP012

### PR DESCRIPTION
**Pull request by jguillen15**
Create draft and an example for BP012, on how to annotate a score with an identity and timestamp in SBOL

### CODE REVIEW

**Review by tcmitchell — CHANGES_REQUESTED** [review]
Hi @jguillen15, thanks very much for submitting this PR.

A bunch of files were added to this PR that should not be present. There is a new `.idea` folder at the top level with a number of files and a subdirectory. Please remove these.

There is also an `.ipynb_checkpoints` directory inside of `SBOL/best-practices/BP012` that I think should also be deleted. If I am mistaken, please let me know.

In summary, I think there should only be two files in this PR:
* `SBOL/best-practices/BP012/README.md`
* `SBOL/best-practices/BP012/example_score_annotation_SBOL.ipynb`

**Comment by jguillen15** [comment]
Hi @tcmitchell ! Sorry , I did not mean to add those extra files, not sure why those were added, I think I fixed it, let me know otherwise, thank you!

**Comment by tcmitchell** [comment]
Hi @jguillen15, thanks again for your submission. We discussed this today at the SBOL Editors meeting. We think you're on the right track, and we'd like to see a few improvements if you don't mind. We think that the BP could be generalized a bit more, and elaborated on, to discuss annotating sequences, or other SBOL classes, with metrics, not just scores. The technique you suggest is a good use of SBOL, and we'd like to see it broadened out.

We think it would also be good to include a picture. This might be a UML diagram, like the one on [BP 10](https://github.com/SynBioDex/SBOL-examples/tree/main/SBOL/best-practices/BP010) or a diagram like the ones at https://www.w3.org/2018/09/rdf-data-viz/.

Additionally, would you be able to add a second example in the text, not necessarily in a notebook, to get a better sense that this is truly a best practice with broader applicability than the iGEM sequence complexity score?

Please let us know your thoughts, or if you need help.

**Comment by jguillen15** [comment]
Hi @tcmitchell , thanks for taking the time for reviewing the proposal.

I will try my best to incorporate these improvements that you talked about.

Is there a specific software I can use to make the UML diagrams?

Thank you

**Comment by jakebeal** [comment]
@jguillen15 We've often used Umlet: https://www.umlet.com/

**Comment by jguillen15** [comment]
Thanks @jakebeal !

**Comment by jguillen15** [comment]
Hi @tcmitchell , sorry for the delay in my response.

I added the UML diagram of the proposed BP and also changed a little bit its description.

I talked to my team and decided that we don't intend to propose a generalized method for annotating all possible scores or metrics in SBOL, for now our purpose is only to record a way in which we could store a complexity score(not necessarily from iGEM) 
I understand that the procedure could be broadened out, but for now it is out of the scope for us.

Please let me know your thoughts.

Thank you!

**Comment by tcmitchell** [comment]
@jguillen15 this looks great to me. There is one last thing to do. Please add your new best practice to the list in `SBOL/best-practices/README.md` so that others will find it more easily.

Thanks!

**Comment by jguillen15** [comment]
Hi @tcmitchell , thanks!

I added BP012 in SBOL/best-practices/README.md.

Let me know if there is anything else to do.

Thank you.
